app/routes.py

```diff
@@ -1,2 +1,257 @@
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            return {"details": "Invalid data"}, 400
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body["completed_at"])
+        db.session.add(new_task)
+        db.session.commit()
+        new_task_response = {
+            "task": {
+                "id": new_task.task_id,
+                "title": new_task.title,
+                "description": new_task.description,
+                "is_complete": False if new_task.completed_at == None else True
+            }
+        }
+        return jsonify(new_task_response), 201
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+
+    if request.method == "GET":
+        if task.goal_id is None:
+            return {
+                "task": {
+                    "id": task.task_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False if task.completed_at == None else True
+                }
+            }
+        if task.goal_id is not None:
+            return {
+                "task": {
+                    "id": task.task_id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": False if task.completed_at == None else True
+                }
+            }
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        db.session.commit()
+        return {
+            "task": {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": False if task.completed_at == None else True
+            }
+        }
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return {
+            "details": f'Task {task.task_id} \"{task.title}\" successfully deleted'
+        }
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def task_complete(task_id):
+    task = Task.query.get(task_id)
+    if task is None:
+        return make_response("", 404)
+
+    if request.method == "PATCH":
+        task.completed_at = datetime.now()
+        db.session.commit()
+        headers = {"Authorization": f"Bearer {token}"}
+        data = {
+            "channel": "task-notifications",
+            "text": "Someone just completed the task My Beautiful Task"
+        }
+        requests.post(path, headers=headers, data=data)
```

Comment:
We can refactor this function so that all the logic pertaining to posting on slack would be in a separate function.